File: 1877-minimize-maximum-pair-sum-in-array/1877-minimize-maximum-pair-sum-in-array.py
class Solution:
    def minPairSum(self, nums: List[int]) -> int:
        nums.sort()
        left = 0
        right = len(nums)-1
        ans = -float('inf')
        while left<right:
            ans = max(ans,nums[left]+nums[right])
            left+=1
            right-=1
        return ans

File: 1877-minimize-maximum-pair-sum-in-array/1877-minimize-maximum-pair-sum-in-array.cpp
class Solution {
public:
    // time/space: O(nlogn)/O(1)
    int minPairSum(vector<int>& nums) {
        int n = nums.size();
        sort(nums.begin(), nums.end());

        int max_sum = INT_MIN;
        for (int i = 0, j = n - 1; i < j; i++, j--) {
            max_sum = max(max_sum, nums[i] + nums[j]);
        }
        return max_sum;
    }
};